### PR TITLE
feat: ExpressionDimensionItem caching strategy [DHIS2-14083-6]

### PR DESCRIPTION
**Pull request by d-bernat**
please see https://dhis2.atlassian.net/browse/DHIS2-14083
and https://dhis2.slack.com/archives/C04CN39F7C7/p1670236817944369
+
ExpressionDimensionItemStore interface is not needed and was removed

### CODE REVIEW

**Review by maikelarabori — APPROVED** [review]

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/DataQueryParams.java [hunk 1]
```diff
@@ -690,6 +691,8 @@ protected QueryKey getQueryKey()
             .add( "order", order )
             .add( "timeField", timeField )
             .add( "orgUnitField", orgUnitField )
+            .add( "expressiondimensionitems", this.getExpressionDimensionItems().stream()
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/DataQueryParams.java:694]: That's good. Could this deserve its own method, like `getExpressionDimensionItemExpressionString` to make the code more readable?

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/DataQueryParams.java [hunk 2]
```diff
@@ -690,6 +691,8 @@ protected QueryKey getQueryKey()
...
+                .map( edi -> ((ExpressionDimensionItem) edi).getExpression() ).collect( Collectors.joining() ) )
             .addIgnoreNull( "apiVersion", apiVersion )
             .addIgnoreNull( "locale", locale );
```
Comment (larshelge) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/DataQueryParams.java:697]: Would be nice with another test in `QueryKeyTest` to validate that the hash key is different based on different EDIs.

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/DataQueryParams.java (hunk 1 above)
Comment (d-bernat) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/DataQueryParams.java:694]: Done

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/DataQueryParams.java (hunk 2 above)
Comment (d-bernat) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/DataQueryParams.java:697]: Done

**Review by gnespolino — APPROVED** [review]

**Pull request merged**